### PR TITLE
Fixed problem with results not showing in presentations.

### LINKED ISSUE #1456: [Bug] Presentations are not showing results from experiment
### Summary

When creating a presentation in GrowthBook, the presentations do not show the results from the selected experiment(s) in the presentation.

The console logs:
```
Uncaught (in promise) Error: Could not establish connection. Receiving end does not exist. /present/pres_<presentation_id>:1
```

While the network shows a successful request to fetch the experiments for the presentation: `GET presentation/pres_<presenation_id>`

See screenshot below for an example on how it looks in the presentation:
Screenshot 2023-07-13 at 10 00 27

#### Expected Behavior

When looking at the presentation, slide 3 and onwards should show the result of the experiment. 

#### Current Behavior

When looking at the presentation, slide 3 and onwards shows an empty result where it also shows an incorrect metric (possibly a default set in GrowthBook) of `Users`.

#### Slack thread

See Slack thread for more context: https://growthbookusers.slack.com/archives/C01T6PKD9C3/p1689235726447139

### PR DESCRIPTION
**Pull request by Auz**
Fixed an error where results are not shown in presentations due to a change in the data structure for snapshot results. Also fixed errors related to possibly undefined variables. 
Fixes #1456

### CODE REVIEW

File: packages/front-end/components/Share/Presentation.tsx [hunk 1]
```diff
@@ -75,51 +76,40 @@ const Presentation = ({
     const e = em.get(eid);
```
Comment (msamper) [inline packages/front-end/components/Share/Presentation.tsx:77]:
```suggestion
if (!e) return;
```

**Review by lukesonnet — APPROVED** [review]

File: packages/front-end/components/Share/Presentation.tsx [hunk 2]
```diff
@@ -73,53 +74,41 @@ const Presentation = ({
...
-
     // get the info on which variation to mark as winner/loser
-    const variationExtra = [];
+    const variationExtra: JSX.Element[] = [];
     let sideExtra = <></>;
     const variationsPlural =
-      // @ts-expect-error TS(2532) If you come across this, please fix it!: Object is possibly 'undefined'.
-      e.experiment.variations.length > 2 ? "variations" : "variation";
+      (e?.experiment?.variations?.length || 0) > 2 ? "variations" : "variation";
```
Comment (bttf) [inline packages/front-end/components/Share/Presentation.tsx:81]: minor copy note ('0 variations' vs '0 variation')
```suggestion
      (e?.experiment?.variations?.length || 0) !== 1 ? "variations" : "variation";
```

File: packages/front-end/components/Share/Presentation.tsx [hunk 3]
```diff
@@ -160,31 +146,50 @@ const Presentation = ({
...
+                {v?.screenshots[0]?.path && (
+                  <img
+                    className="expimage border"
+                    src={v.screenshots[0].path}
+                    alt={v.name}
+                  />
+                )}
+                {v.description && (
+                  <div
+                    className={`text-center`}
```
Comment (bttf) [inline packages/front-end/components/Share/Presentation.tsx:187]: minor syntax note
```suggestion
                    className="text-center"
```

File: packages/front-end/components/Share/ShareModal.tsx [hunk 4]
```diff
@@ -263,8 +263,7 @@ const ShareModal = ({
       }
       if (onSuccess && typeof onSuccess === "function") onSuccess();
       setLoading(false);
-      // @ts-expect-error TS(2722) If you come across this, please fix it!: Cannot invoke an object which is possibly 'undefin... Remove this comment to see the full error message
-      refreshList();
+      if (refreshList) refreshList();
```
Comment (bttf) [inline packages/front-end/components/Share/ShareModal.tsx:266]: FYI - you could also do `refreshList?.()`

**Review by bttf — APPROVED** [review]
Looks good to me but maybe a second review might do well

File: packages/front-end/components/Share/Presentation.tsx [hunk 5]
```diff
@@ -160,31 +146,50 @@ const Presentation = ({
...
+          </Heading>
+          <Text className="text-center m-0 mb-4 p-2 px-5" fontSize={21}>
+            {e?.experiment?.hypothesis
+              ? "Hypothesis: " + e.experiment.hypothesis
+              : ""}
+            {e?.experiment?.metrics && (
+              <>
+                <br />
+                <span style={{ fontSize: "0.95rem" }}>
+                  Primary metrics: {e?.experiment?.metrics.join(", ")}
```
Comment (lukesonnet) [inline packages/front-end/components/Share/Presentation.tsx:162]: These are the metric IDs and therefore are not informative (all you see is `met_***, met_***`). Consider getting the metrics right from the snapshot itself.

**Review by lukesonnet — CHANGES_REQUESTED** [review]

File: packages/front-end/components/Share/Presentation.tsx (hunk 5 above)
Comment (Auz) [reply packages/front-end/components/Share/Presentation.tsx]: fixed

File: packages/front-end/components/Share/ShareModal.tsx (hunk 4 above)
Comment (Auz) [reply packages/front-end/components/Share/ShareModal.tsx]: Done

**Pull request merged**